doc/blog/Wasm-JS series/Blog - WASM.md

```diff
@@ -0,0 +1,672 @@
+               let html = code;
+               if(prism.languages[language]) {{
+                   // When the specified language is supported by PrismJS...
+                   html = prism.highlight(code, prism.languages[language], language);
+               }}
+   
+               // Get HTML element
+               const element = document.getElementById(""{HtmlId}"");
+               if(!element) return; // Not in DOM yet
+   
+               // Display result
+               element.innerHTML = html;
+   
+               // Set CSS classes, when required
+               if(oldLanguageCss) {{
+                   element.classList.remove(oldLanguageCss);
+               }}
+               if(newLanguageCss) {{
+                   element.classList.add(newLanguageCss);
+               }}
+           }})();";
+   
+       WebAssemblyRuntime.InvokeJS(javascript);
+   }
+   ```
+
+3. Change `CodeChanged()` and `LanguageChanged()` to call the new `UpdateDisplay()` method:
+
+   ``` csharp
+   private static void CodeChanged(DependencyObject dependencyobject, DependencyPropertyChangedEventArgs args)
+   {
+       (dependencyobject as PrismJsView)?.UpdateDisplay();
+   }
+   
+   private static void LanguageChanged(DependencyObject dependencyobject, DependencyPropertyChangedEventArgs args)
+   {
+       (dependencyobject as PrismJsView)?.UpdateDisplay(args.OldValue as string, args.NewValue as string);
+   }
+   ```
+
+4. We also need to update the result when the control is loaded in the DOM. So we need to change the constructor again like this:
+
+   ``` csharp
+   public PrismJsView() : base("code") // PrismJS requires a <code> element
+   {
+       // This is required to set to <code> style for PrismJS to works well
+       // https://github.com/PrismJS/prism/issues/1237#issuecomment-369846817
+       SetStyle("white-space", "pre-wrap");
+   
+       // Update the display when the element is loaded in the DOM
+       Loaded += (snd, evt) => UpdateDisplay(newLanguage: Language);
+   }
+   ```
+
+5. Compile & run.  It should work like this:
+   ![image-20200415135422628](image-20200415135422628.png)
+
+## 🔬 Going further
+
+That's a very simple integration because there's no _callback_ from HTML to managed code and _PrismJS_ is a self-contained framework.
+
+Few improvements could be done to make the code more _production ready_:
+
+* **Make the control multi-platform**. The simple way would be to use a WebView on other platforms: that way the exact same text-rendering framework would be used everywhere. Right now this code won't compile on other targets.
+* **Create script files instead of generating dynamic javascript**. That would have the advantage of improving performance and increase the ability to debug it. Few projects are also using TypeScript to generate javascript. This approach is done by Uno itself for the `Uno.UI.Wasm` project: https://github.com/unoplatform/uno/tree/master/src/Uno.UI.Wasm.
```

Comment:
```suggestion
* **Create script files instead of generating dynamic Javascript**. That would have the advantage of improving performance and make it easier to debug the code. A few projects are also using TypeScript to generate Javascript. This approach is done by Uno itself for the `Uno.UI.Wasm` project: https://github.com/unoplatform/uno/tree/master/src/Uno.UI.Wasm.
```